lib/pronto/formatter/text_formatter.rb

```diff
@@ -14,9 +30,30 @@ def location(message)
         line = message.line
         lineno = line.new_lineno if line
         path = message.path
-        commit_sha = message.commit_sha[0..6] if message.commit_sha
+        commit_sha = message.commit_sha
+
+        if path || lineno
```

Comment:
You've introduced mixed levels of abstraction here.

`colorize(...)` is called in a method named `level` but also in methods named `format_location_*`. Logic around what to display is mixed between `location` and `format_location_*`.

I would suggest reducing it to only two layers, `format` which calls `format_level` and `format_location`. Inline `format_location_path` and `format_location_commit` into `format_location`, all they do now is add another layer of indirection.

Does that make sense? It's definitely not bad now, it would work as it. I just think it could have a little bit more unity in how things are laid out.